tests/test-rest-posts-controller.php

```diff
@@ -455,6 +457,20 @@ public function test_get_items_private_filter_query_var() {
 		$this->assertEquals( $draft_id, $data[0]['id'] );
 	}
 
+	public function test_get_items_invalid_per_page() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array( 'per_page' => -1 ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );
+	}
+
+	public function test_get_items_invalid_filter_posts_per_page() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array( 'filter' => array( 'posts_per_page' => -1 ) ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );
+	}
```

Comment:
This test is failing because it either represents an error condition we don't account for, or because it's the wrong test; hence the Q in the description